### PR TITLE
MNT Travis shared config

### PR DESCRIPTION
**Pull request by emteknetnz**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,40 +1,51 @@
...
+        - DB=MYSQL
+        #- PHPUNIT_COVERAGE_TEST=1
+        - PHPUNIT_TEST=1
+    - php: nightly
+      env:
+        - DB=MYSQL
+        - PHPUNIT_TEST=1
+        - COMPOSER_INSTALL_ARG=--ignore-platform-reqs
+  allow_failures:
     - php: nightly
```
Comment (Cheddam) [inline .travis.yml:43]: We don't want to allow failures against PHP 8 in core.

File: .travis.yml [hunk 2]
```diff
@@ -1,40 +1,51 @@
...
-  - ~/.config/composer/vendor/bin/cow schema:validate
-
-script:
-  - vendor/bin/phpunit --testsuite recipe-core
+  # marcj/topsort:^2@dev is a temporary requirement for PHP8 support
+  # This requirement haven't been shipped stable yet
+  # When it is stable then composer.json for silverstripe/config should be updated to ^2
+  # and this line should be removed
+  # ^1 is included because ^2@dev has a minimum php requirement of PHP 7.3
+  - if [ $(php -r 'echo (int) version_compare(phpversion(), "8.0.0", ">=");') = "1" ] ; then compose require marcj/topsort:^2@dev; fi
```
Comment (Cheddam) [inline .travis.yml:51]: This can be dropped now, as `marcj/topsort ^2` has shipped a stable release.

File: .travis.yml [hunk 3]
```diff
@@ -1,40 +1,51 @@
...
+    - REQUIRE_EXTRA="silverstripe-themes/simple:~3.2.0 silverstripe/admin:1.7.x-dev silverstripe/versioned:1.7.x-dev mikey179/vfsstream:^1.6"
+    - PHPUNIT_SUITE="recipe-core"
+
+# Using a manual matrix to exlude PHPUNIT_COVERAGE_TEST which was having an unknown error
+# https://travis-ci.com/github/silverstripe/recipe-core/jobs/457587137
+jobs:
+  fast_finish: true
   include:
     - php: 7.1
-      env: DB=MYSQL COMPOSER_ARG=--prefer-lowest
```
Comment (Cheddam) [inline .travis.yml:16]: We should look at introducing a `--prefer-lowest` build to the shared matrix - it's a useful testing target. For now, it'd be good to retain it here.

**Review by Cheddam — CHANGES_REQUESTED** [review]
Nice work, just a couple of items to address.

File: .travis.yml (hunk 3 above)
Comment (emteknetnz) [reply .travis.yml:16]: Have added it back in

File: .travis.yml (hunk 2 above)
Comment (emteknetnz) [reply .travis.yml]: Have removed

File: .travis.yml (hunk 1 above)
Comment (emteknetnz) [reply .travis.yml]: Have removed this bit

**Review by Cheddam — APPROVED** [review]

**Pull request merged**